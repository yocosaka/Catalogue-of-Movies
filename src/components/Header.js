import React from 'react';

const Header = () => (
  <header className="header">
    <h1 className="header__title">Catalogue of Recipes</h1>
  </header>
);

export default Header;
