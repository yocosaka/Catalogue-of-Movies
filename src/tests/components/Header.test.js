import React from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../../components/Header';

describe('Header with Link', () => {
  test('full app rendering/navigating', () => {
    render(<Header />, { wrapper: MemoryRouter });
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('MoviSearch')).toBeInTheDocument;
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
