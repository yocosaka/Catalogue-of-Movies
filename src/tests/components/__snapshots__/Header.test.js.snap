// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Link should match with snapshot 1`] = `
<header
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="header__title"
    >
      MoviSearch
    </h1>
  </a>
</header>
`;
